sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -521,35 +521,114 @@ case class AlterTableRenamePartitionCommand(
  */
 case class AlterTableDropPartitionCommand(
```

Comment:
Shall we make table relation as a child? then we can resolve the `partitionsFilters` automatically.

Reply:
I thought about that. The point is that we have anyway to check that the attributes specified are the partitioning ones. So I am not sure it is worth to run the whole analyzer rules for something we have anyway to handle somehow.

Reply:
But it's also weird to use `AttributeReference` this way. Can we create a new `Attribute` implementation for this purpose? Basically we only need a resolved expression to hold the partition column name. The type doesn't matter.

Reply:
sure, I will. Thanks.